src/main/Core/ImageGenerator/LinuxIconExtractor.ts

```diff
@@ -0,0 +1,412 @@
+import type { CommandlineUtility } from "@Core/CommandlineUtility";
+import type { FileSystemUtility } from "@Core/FileSystemUtility";
+import type { IniFileParser } from "@Core/IniFileParser/IniFileParser";
+import type { Logger } from "@Core/Logger";
+import type { Image } from "@common/Core/Image";
+import { dirname, extname, join } from "path";
+import type { CacheFileNameGenerator } from "./CacheFileNameGenerator";
+import type { FileIconExtractor } from "./FileIconExtractor";
+
+// Specs from https://specifications.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html
+type IconThemeSubdir = {
+    Size: number;
+    Scale: number; // Defaults to 1
+    MinSize: number; // Defaults to size
+    MaxSize: number; // Defaults to size
+    Threshold: number; // Defaults to 2
+    Type: "Fixed" | "Scalable" | "Threshold"; // Defaults to Threshold
+};
+
+type IconTheme = {
+    name: string;
+    path: string;
+    // Paths in order as listed in index.theme
+    subdirectories: string[];
+    // Map<subdirPath, subdirData>
+    subdirData: Map<string, IconThemeSubdir>;
+    parents: string[];
+};
+
+export class LinuxIconExtractor implements FileIconExtractor {
```

Comment:
And is there a way to reduce the size of this class? It's already quite big, which is an indicator that it does a lot of different things. This makes it really hard to write maintainable tests. I suggest finding a way to extract some code of this class into other classes or utilities (or just functions).

Reply:
I can try to pull some of the code into a second class, but the reason the class is so large is because finding icons on Linux is a fairly complex process due to the ability to theme their own desktops